tests/integration_tests/scheduler/test_slurm_driver.py

```diff
@@ -0,0 +1,184 @@
+
+from ert.scheduler import SlurmDriver
+from tests.utils import poll
+
+from .conftest import mock_bin
+
+
+@pytest.fixture(autouse=True)
+def mock_slurm(pytestconfig, monkeypatch, tmp_path):
+    if pytestconfig.getoption("slurm"):
+        # User provided --slurm, which means we should use an actual Slurm
+        # cluster without mocking anything.""
+        return
+    mock_bin(monkeypatch, tmp_path)
+
+
+async def test_slurm_stdout_file(tmp_path, job_name):
+    os.chdir(tmp_path)
+    driver = SlurmDriver()
+    await driver.submit(0, "sh", "-c", "echo yay", name=job_name)
+    await poll(driver, {0})
+    slurm_stdout = Path(f"{job_name}.stdout").read_text(encoding="utf-8")
+    assert Path(f"{job_name}.stdout").exists(), "Slurm system did not write output file"
+    assert "yay" in slurm_stdout
+
+
+async def test_slurm_dumps_stderr_to_file(tmp_path, job_name):
+    os.chdir(tmp_path)
+    driver = SlurmDriver()
+    failure_message = "failURE"
+    await driver.submit(0, "sh", "-c", f"echo {failure_message} >&2", name=job_name)
+    await poll(driver, {0})
+    assert Path(f"{job_name}.stderr").exists(), "Slurm system did not write stderr file"
+
+    assert (
+        Path(f"{job_name}.stderr").read_text(encoding="utf-8").strip()
+        == failure_message
+    )
+
+
+def generate_random_text(size):
+    letters = string.ascii_letters
+    return "".join(random.choice(letters) for i in range(size))
+
+
+@pytest.mark.parametrize("tail_chars_to_read", [(5), (50), (500), (700)])
+async def test_slurm_can_retrieve_stdout_and_stderr(
+    tmp_path, job_name, tail_chars_to_read
+):
+    os.chdir(tmp_path)
+    driver = SlurmDriver()
+    num_written_characters = 600
+    _out = generate_random_text(num_written_characters)
+    _err = generate_random_text(num_written_characters)
+    await driver.submit(0, "sh", "-c", f"echo {_out} && echo {_err} >&2", name=job_name)
+    await poll(driver, {0})
+    message = driver.read_stdout_and_stderr_files(
+        runpath=".",
+        job_name=job_name,
+        num_characters_to_read_from_end=tail_chars_to_read,
+    )
+
+    stderr_txt = Path(f"{job_name}.stderr").read_text(encoding="utf-8").strip()
+    stdout_txt = Path(f"{job_name}.stdout").read_text(encoding="utf-8").strip()
+
+    assert stderr_txt[-min(tail_chars_to_read, num_written_characters) + 2 :] in message
+    assert stdout_txt[-min(tail_chars_to_read, num_written_characters) + 2 :] in message
+
+
+@pytest.mark.integration_test
+async def test_submit_to_named_queue(tmp_path, caplog, job_name):
+    """If the environment variable _ERT_TEST_ALTERNATIVE_QUEUE is defined
+    a job will be attempted submitted to that queue.
+
+    * Note that what is called a "queue" in Ert is a "partition" in Slurm lingo.
+
+    As Ert does not keep track of which queue a job is executed in, we can only
+    test for success for the job."""
+    os.chdir(tmp_path)
+    driver = SlurmDriver(queue_name=os.getenv("_ERT_TESTS_ALTERNATIVE_QUEUE"))
```

Comment:
When submitting to the real queue, should we introduce some grace waiting / sleep times?

Reply:
We don't even have that for the LSF tests. The entire thing will timeout anyhow. 

When we run Slurm integration tests, it is on Github actions with an ad-hoc local Slurm cluster which always will respond immediately, as there are no other users.